src/requests/adapters.py

```diff
@@ -304,8 +332,10 @@ def cert_verify(self, conn, url, verify, cert):
                     f"Could not find a suitable TLS CA certificate bundle, "
                     f"invalid path: {cert_loc}"
                 )
-
-            conn.cert_reqs = "CERT_REQUIRED"
+            
+            if is_single_certificate(cert_loc):
```

Comment:
Doesn't this just disable certificate verification if there is a single cert in the chain, without actually verifying that the cert provided by the server matches?